dotcom-rendering/src/web/components/EmailSignup.tsx

```diff
@@ -0,0 +1,132 @@
+import { css } from '@emotion/react';
+import {
+	brandAltBackground,
+	from,
+	headline,
+	space,
+	textSans,
+} from '@guardian/source-foundations';
+import { SvgEnvelope } from '@guardian/source-react-components';
+import React from 'react';
+import { SecureSignup } from './SecureSignup';
+
+type Props = {
+	newsletterId: string;
+	name: string;
+	description: string;
+	frequency: string;
+	successText: string;
+};
+
+const Container = ({ children }: { children: React.ReactNode }) => {
+	return (
+		<div
+			css={css`
+				border: black 3px dashed;
+				border-radius: 12px;
+				padding: 10px;
+				margin-bottom: ${space[3]}px;
+			`}
+		>
+			{children}
+		</div>
+	);
+};
+
+const Title = ({ name }: { name: string }) => {
+	return (
+		<div
+			css={css`
+				${headline.xsmall({ fontWeight: 'bold' })}
+				flex-grow: 1;
+			`}
+		>
+			Sign up to {name} today
+		</div>
+	);
+};
+
+const Frequency = ({ frequency }: { frequency: string }) => {
+	return (
+		<div
+			css={css`
+				display: flex;
+				margin-top: ${space[2]}px;
+				${from.tablet} {
+					margin-top: 0;
+				}
+			`}
+		>
+			<div
+				css={css`
+					svg {
+						background: ${brandAltBackground.primary};
+						border-radius: 50%;
+						height: 28px;
+						padding: 2px;
+						margin-right: ${space[1]}px;
+					}
+				`}
+			>
+				<SvgEnvelope />
+			</div>
+			<div
+				css={css`
+					${textSans.medium({ fontWeight: 'bold' })}
+				`}
+			>
+				{frequency}
+			</div>
+		</div>
+	);
+};
+
+const Description = ({ description }: { description: string }) => {
+	return (
+		<div
+			css={css`
+				${textSans.medium({ lineHeight: 'tight' })}
+				margin-top: ${space[1]}px;
+				margin-bottom: ${space[2]}px;
+			`}
+		>
+			{description}
+		</div>
+	);
+};
+
+const StackBelowTablet = ({ children }: { children: React.ReactNode }) => (
+	<div
+		css={css`
+			display: flex;
+			flex-direction: column;
+			${from.tablet} {
+				flex-direction: row;
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+
+export const EmailSignup = ({
+	newsletterId,
+	name,
+	description,
+	frequency,
+	successText,
+}: Props) => {
+	return (
+		<Container>
+			<StackBelowTablet>
+				<Title name={name} />
+				<Frequency frequency={frequency} />
+			</StackBelowTablet>
+			<Description description={description} />
+			<SecureSignup
+				newsletterId={newsletterId}
+				successText={successText}
+			/>
+		</Container>
```

Comment:
I think this probably comes entirely down to a matter of personal opinion, but a lot of these components created in this file are only used once, and I wonder if it would be more readable to keep all the html in one place; then extract the CSS from being inline to declared in variables, e.g

```jsx
const containerStyles = css`
	border: black 3px dashed;
	border-radius: 12px;
	padding: 10px;
	margin-bottom: ${space[3]}px;
`

const stackBelowTabletStyles = ...

const titleStyles = ...

return (
    <div css={containerStyles}>
		<div css={stackBelowTabletStyles}>
			<div css={titleStyles}>{title}</div>
             ....
		</div>
		...
	</div>
)

```

I think this is more fitting on the 'style' of DCR (e.g [CricketScoreboard.tsx](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/components/CricketScoreboard.tsx)) - but its definitely subjective!

Reply:
Yes, that works nicer here. I think it would make sense to extract the 'Frequency' component to its own file (its a recurring design element that will be used in other newsletter-y components) and use CSS variables for the rest.